src/Streamly/Internal/Data/Stream/Serial.hs

```diff
@@ -213,15 +214,19 @@ map f = mapM (return . f)
 
 {-# INLINE apSerial #-}
 apSerial :: Monad m => SerialT m (a -> b) -> SerialT m a -> SerialT m b
-apSerial (SerialT m1) (SerialT m2) =
-    let f x1 = D.concatMapU (UF.singleton (pure . x1)) (D.toStreamD m2)
-    in D.fromStreamD $ D.concatMap f (D.toStreamD m1)
+apSerial (SerialT m1) (SerialT m2) = D.fromStreamD $ D.toStreamD m1 <*> D.toStreamD m2
```

Comment:
Use `D.concatAp` and `D.apSequence` directly instead of depending on the `Applicative` type class instance resolution?